lib/registries/addon/drafts/draft/-components/register/component.ts

```diff
@@ -7,48 +7,59 @@ import { run } from '@ember/runloop';
 import { inject as service } from '@ember/service';
 import { task } from 'ember-concurrency-decorators';
 import DS from 'ember-data';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
 
 import { tracked } from '@glimmer/tracking';
 import DraftRegistration from 'ember-osf-web/models/draft-registration';
 import NodeModel from 'ember-osf-web/models/node';
 import Registration from 'ember-osf-web/models/registration';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
 import DraftRegistrationManager from 'registries/drafts/draft/draft-registration-manager';
 
 @tagName('')
 export default class Register extends Component {
     @service store!: DS.Store;
+    @service toast!: Toast;
+    @service intl!: Intl;
 
     // Required
     draftManager!: DraftRegistrationManager;
 
     // Private
-    @tracked registration!: Registration;
     onSubmitRedirect?: (registrationId: string) => void;
     @alias('draftManager.hasInvalidResponses') isInvalid?: boolean;
     @alias('draftManager.draftRegistration') draftRegistration!: DraftRegistration;
     @alias('draftManager.hasProject') hasProject!: boolean;
     @alias('draftManager.node') node?: NodeModel;
     @alias('draftManager.currentUserIsAdmin') currentUserIsAdmin!: boolean;
 
+    @tracked registration!: Registration;
     @tracked partialRegDialogIsOpen = false;
     @tracked finalizeRegDialogIsOpen = false;
 
     @task({ withTestWaiter: true })
     onClickRegister = task(function *(this: Register) {
         if (!this.registration) {
-            const registration = this.store.createRecord('registration', {
+            this.registration = this.store.createRecord('registration', {
                 draftRegistrationId: this.draftRegistration.id,
-                registeredFrom: this.hasProject ? this.draftRegistration.branchedFrom : null,
                 provider: this.draftRegistration.provider,
+                ...(this.hasProject ? { registeredFrom: this.draftRegistration.branchedFrom } : {}),
             });
-
-            this.setProperties({ registration });
         }
+
         if (this.node && this.hasProject) {
-            yield this.node.loadRelatedCount('children');
-        }
-        if (this.hasProject && this.node && this.node.relatedCounts.children > 0) {
-            this.showPartialRegDialog();
+            try {
+                yield this.node.loadRelatedCount('children');
+            } catch (e) {
+                const errorMessage = this.intl.t('registries.drafts.draft.unable_to_fetch_children_count');
```

Comment:
Did this come up as part of the changes above, or was this an opportunistic catch?

Reply:
Opportunistic catch. We should wrap these async calls in a try catch and maybe send to sentry. The message language can change if need be.